generator/main.go

```diff
@@ -0,0 +1,485 @@
+		Spec: v1beta1.CommunitySpec{
+			Communities: communitiesAliases,
+		},
+	}
+	return []v1beta1.Community{ret}
+}
+
+func peersFor(c *configFile) ([]v1beta2.BGPPeer, error) {
+	var ret []v1beta2.BGPPeer
+	for i, peer := range c.Peers {
+		p, err := parsePeer(peer)
+		if err != nil {
+			return nil, err
+		}
+		p.Name = fmt.Sprintf("peer%d", i+1)
+		p.Namespace = resourcesNameSpace
+		ret = append(ret, *p)
+	}
+	return ret, nil
+}
+
+func parsePeer(p peer) (*v1beta2.BGPPeer, error) {
+	holdTime, err := parseHoldTime(p.HoldTime)
+	if err != nil {
+		return nil, err
+	}
+	keepaliveTime, err := parseKeepaliveTime(holdTime, p.KeepaliveTime)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeSels := make([]metav1.LabelSelector, 0)
+	for _, sel := range p.NodeSelectors {
+		s := parseNodeSelector(sel)
+		nodeSels = append(nodeSels, *s)
+	}
+
+	return &v1beta2.BGPPeer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: resourcesNameSpace,
+		},
+		Spec: v1beta2.BGPPeerSpec{
+			MyASN:         p.MyASN,
+			ASN:           p.ASN,
+			Address:       p.Addr,
+			SrcAddress:    p.SrcAddr,
+			Port:          p.Port,
+			HoldTime:      metav1.Duration{Duration: holdTime},
+			KeepaliveTime: metav1.Duration{Duration: keepaliveTime},
+			RouterID:      p.RouterID,
+			NodeSelectors: nodeSels,
+			Password:      p.Password,
+			BFDProfile:    p.BFDProfile,
+			EBGPMultiHop:  p.EBGPMultiHop,
+		},
+	}, nil
+}
+
+func parseNodeSelector(sel nodeSelector) *metav1.LabelSelector {
+	ret := &metav1.LabelSelector{}
+
+	ret.MatchLabels = sel.MatchLabels
+	ret.MatchExpressions = []metav1.LabelSelectorRequirement{}
+
+	for _, m := range sel.MatchExpressions {
+
+		matchExp := &metav1.LabelSelectorRequirement{
+			Key:      m.Key,
+			Operator: metav1.LabelSelectorOperator(m.Operator),
+			Values:   m.Values,
+		}
+		matchExp.Values = make([]string, len(m.Values))
+		copy(matchExp.Values, m.Values)
+		ret.MatchExpressions = append(ret.MatchExpressions, *matchExp)
+	}
+	return ret
+}
+
+func parseHoldTime(ht string) (time.Duration, error) {
+	if ht == "" {
+		return 90 * time.Second, nil
+	}
+	d, err := time.ParseDuration(ht)
+	if err != nil {
+		return 0, fmt.Errorf("invalid hold time %q: %s", ht, err)
+	}
+	rounded := time.Duration(int(d.Seconds())) * time.Second
+	if rounded != 0 && rounded < 3*time.Second {
+		return 0, fmt.Errorf("invalid hold time %q: must be 0 or >=3s", ht)
+	}
+	return rounded, nil
+}
+
+func parseKeepaliveTime(ht time.Duration, ka string) (time.Duration, error) {
+	// If keepalive time not set lets use 1/3 of holdtime.
+	if ka == "" {
+		return ht / 3, nil
+	}
+	d, err := time.ParseDuration(ka)
+	if err != nil {
+		return 0, fmt.Errorf("invalid keepalive time %q: %s", ka, err)
+	}
+	rounded := time.Duration(int(d.Seconds())) * time.Second
+	return rounded, nil
+}
+
+func ipAddressPoolsFor(c *configFile) []v1beta1.IPAddressPool {
+	ret := make([]v1beta1.IPAddressPool, len(c.Pools))
+	for i, addresspool := range c.Pools {
+		var ap v1beta1.IPAddressPool
```

Comment:
this has an extra complication:
the configMap has avoidBuggyIPs while we don't. So we need to convert CIDR to ranges excluding the extremes if the flag is set, and to exclude extremes from ranges.